lib/ansible/modules/cloud/vmware/vcenter_folder.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Davis Phillips davis.phillips@gmail.com
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: vcenter_folder
+short_description: Add/remove folders to/from vCenter
+description:
+- This module can be used to add/remove a folder to/from vCenter.
+version_added: '2.4'
+author:
+- Davis Phillips (@dav1x)
+- Dag Wieers (@dagwieers)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- PyVmomi
+options:
+  datacenter:
+    description:
+    - Name of the datacenter to add the host.
+    required: yes
+  cluster:
+    description:
+    - Name of the cluster to add the host.
+    required: yes
+  folder:
+    description:
+    - Folder name to manage.
+    required: yes
+  hostname:
+    description:
+    - ESXi hostname to manage.
+    required: yes
+  username:
+    description:
+    - ESXi username.
+    required: yes
+  password:
+    description:
+    - ESXi password.
+    required: yes
+  state:
+    description:
+    - Add or remove the folder.
+    default: present
+    choices: [ absent, present ]
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+# Create a folder
+- name: Add a folder to vCenter
+  vmware_folder:
+    hostname: vcsa_host
+    username: vcsa_user
+    password: vcsa_pass
+    datacenter: datacenter
+    cluster: cluster
+    folder: folder
```

Comment:
Could you show folder: folder/subfolder/subfolder here for the example so user knows how to create a subfolder?